src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/TagHelperDescriptorExtensions.cs

```diff
@@ -14,14 +13,12 @@ internal static class TagHelperDescriptorExtensions
 {
     public static bool IsAnyComponentDocumentTagHelper(this TagHelperDescriptor tagHelper)
     {
-        return tagHelper.IsComponentTagHelper || tagHelper.Metadata.ContainsKey(ComponentMetadata.SpecialKindKey);
+        return tagHelper.Kind >= TagHelperKind.Component;
```

Comment:
>return tagHelper.Kind >= TagHelperKind.Component; [](http://example.com/codeflow?start=8&length=49)

This requirement might be nice to call out explicitly in the TagHelperKind enum

Reply:
Agreed